src/validator.cpp

```diff
@@ -1635,6 +1708,530 @@ void Validator::ValidatorImpl::validateMathMLElements(const XmlNodePtr &node, co
     }
 }
 
+void Validator::ValidatorImpl::addMathMLIssue(const std::string &description,
+                                              Issue::ReferenceRule referenceRule,
+                                              const ComponentPtr &component)
+{
+    auto issue = Issue::IssueImpl::create();
+
+    issue->mPimpl->setDescription(description);
+    issue->mPimpl->mItem->mPimpl->setMath(component);
+    issue->mPimpl->setReferenceRule(referenceRule);
+
+    addIssue(issue);
+}
+
+bool Validator::ValidatorImpl::hasOneMathmlSibling(const XmlNodePtr &parentNode,
+                                                   const XmlNodePtr &node,
+                                                   const ComponentPtr &component)
+{
+    if (mathmlChildCount(parentNode) != 2) {
+        addMathMLIssue("Math has a '" + node->name() + "' element without a MathML sibling.",
+                       Issue::ReferenceRule::MATH_MATHML,
+                       component);
+
+        return false;
+    }
+
+    return true;
+}
+
+bool Validator::ValidatorImpl::hasAtLeastOneMathmlSibling(const XmlNodePtr &parentNode,
+                                                          const XmlNodePtr &node,
+                                                          const ComponentPtr &component)
+{
+    if (mathmlChildCount(parentNode) < 2) {
+        addMathMLIssue("Math has a '" + node->name() + "' element without at least one MathML sibling.",
+                       Issue::ReferenceRule::MATH_MATHML,
+                       component);
+
+        return false;
+    }
+
+    return true;
+}
+
+bool Validator::ValidatorImpl::hasTwoMathmlSiblings(const XmlNodePtr &parentNode,
+                                                    const XmlNodePtr &node,
+                                                    const ComponentPtr &component)
+{
+    if (mathmlChildCount(parentNode) != 3) {
+        addMathMLIssue("Math has a '" + node->name() + "' element without two MathML siblings.",
```

Comment:
Should be "Math has a xxx element without exactly two MathML siblings."

Reply:
472fdbdd